### DISCUSSION TITLE
software doesn't work when playing game

### DISCUSSION THREAD

**Discussion by shiwangyan** [mulaRahul/keyviz#45] [General]
When I play Genshin Impact, the software doesn't show my buttons. I think it's a bug
But in other case it's fine
v1.0.6

 - Windows 10

**Comment by mulaRahul** [comment]
## 🖥️ Fullscreen Mode
That's probably because the software gets hidden/overlayed by your game (_Steam_) window. The same happened in this issue #37 as well. Try running your game in **minimized** mode (_i.e. not fullscreen_) and hopefully you should be able to see the visualization again.

Keyviz displays/overlays itself on top of other applications but the same doesn't apply to games, video players, etc. which are in full-screen mode, because they get higher priority than keyviz. In most situations, this behavior is fine. As it is targeted more towards tutorials/explainers.

## Then 😏...
Interestingly, I think a new project can be made for displaying keystrokes and **gamepad controls** targeted toward gamers (_with RGB and stuff_). And of-course overlay above full-screened windows as well. What are your thoughts?

**Reply by shiwangyan** [reply]
but I played in window mode, not fullscreen

**Reply by mulaRahul** [reply]
Just like Keyviz, Steam (_hopefully where you're playing the game_) wants to display its window on top of other applications as well and it gets the privilege to perform too. *Short story short, keyviz cannot display above Steam with its current setup.*

Keyviz is set to be always on top of other applications but maybe Steam does something else to get on top (even in windowed mode it is on top as you can see for yourself). I need to dig around for more info on the same.

But as I've mentioned above, 
> keyviz is not targeted toward gamers!

**Reply by shiwangyan** [reply]
ok,fine

**Reply by shoucandanghehe** [reply]
> ## Then 😏...
> Interestingly, I think a new project can be made for displaying keystrokes and **gamepad controls** targeted toward gamers (_with RGB and stuff_). And of-course overlay above full-screened windows as well. What are your thoughts?

I'm very excited about this!
By the way I have some ideas.

- For gamers, it should be possible to customize some specific layouts, so that the set keys remain on the screen at all times
- An option to display `kps` (`Keys Per Second`), which would be useful for music/rhythm or other games with fast keystrokes
- A function to count the number of keystrokes, maybe some gamers want to know how many times they have pressed a key
- Covering the full screen window should be optional. For some games, displaying buttons on the game screen will have some impact on the game.But the window should continue to run in the background and be captured by the live broadcast software.

**Reply by mulaRahul** [reply]
Nice, I'll keep them in mind. Though, I cannot assure you that I'm going to work on the same anytime soon 😓.

**Comment by mulaRahul** [comment]
@shiwangyan, someone on youtube commented that "running the app on administer mode" enables it overlay over fullscreen games/apps. Can you verify the same?

**Reply by CTI-Tim** [reply]
This. I found many apps need keyvis to be ran as admin.
